diff --git a/BUILD.gn b/BUILD.gn
index 19e0d2b5847..dcad4ac684e 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -7232,7 +7232,7 @@ if (v8_monolithic) {
   # Using external startup data would produce separate files.
   assert(!v8_use_external_startup_data,
          "Set `v8_use_external_startup_data = false` for v8_monolithic.")
-  v8_static_library("v8_monolith") {
+  v8_static_library("wee8") {
     deps = [
       ":v8",
       ":v8_libbase",
@@ -7245,7 +7245,7 @@ if (v8_monolithic) {
 }
 
 if (v8_enable_webassembly) {
-  v8_static_library("wee8") {
+  v8_static_library("wee9") {
     deps = [
       ":v8_base",
       ":v8_libbase",
