diff --git a/BUILD.gn b/BUILD.gn
index dd97c4f922c..ed2cfa6cdda 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -6675,14 +6675,14 @@ if (v8_check_header_includes) {
 # Produce a single static library for embedders
 #
 
-if (v8_monolithic) {
+#if (v8_monolithic) {
   assert(!is_component_build,
          "Set `is_component_build = false` for v8_monolithic.")
 
   # Using external startup data would produce separate files.
   assert(!v8_use_external_startup_data,
          "Set `v8_use_external_startup_data = false` for v8_monolithic.")
-  v8_static_library("v8_monolith") {
+  v8_static_library("wee8") {
     deps = [
       ":v8",
       ":v8_libbase",
@@ -6692,10 +6692,10 @@ if (v8_monolithic) {
 
     configs = [ ":internal_config" ]
   }
-}
+#}
 
 if (v8_enable_webassembly) {
-  v8_static_library("wee8") {
+  v8_static_library("wee9") {
     deps = [
       ":v8_base",
       ":v8_libbase",
