diff --git a/BUILD.gn b/BUILD.gn
index dd97c4f922c..f028657f12a 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -6694,13 +6694,13 @@ if (v8_monolithic) {
   }
 }
 
-if (v8_enable_webassembly) {
+# if (v8_enable_webassembly) {
   v8_static_library("wee8") {
     deps = [
       ":v8_base",
       ":v8_libbase",
       ":v8_libplatform",
-      ":v8_shared_internal_headers",
+#      ":v8_shared_internal_headers",
       ":v8_snapshot",
       "//build/win:default_exe_manifest",
     ]
@@ -6712,15 +6712,15 @@ if (v8_enable_webassembly) {
 
     configs = [ ":internal_config" ]
 
-    sources = [
-      ### gcmole(all) ###
-      "src/wasm/c-api.cc",
-      "src/wasm/c-api.h",
-      "third_party/wasm-api/wasm.h",
-      "third_party/wasm-api/wasm.hh",
-    ]
+#    sources = [
+#      ### gcmole(all) ###
+#      "src/wasm/c-api.cc",
+#      "src/wasm/c-api.h",
+#      "third_party/wasm-api/wasm.h",
+#      "third_party/wasm-api/wasm.hh",
+#    ]
   }
-}
+# }
 
 ###############################################################################
 # Executables
