diff --git a/BUILD.gn b/BUILD.gn
index 378bd0d..399cfcb 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -69,7 +69,7 @@ source_set("zlib_common_headers") {
 
 use_arm_neon_optimizations = false
 if ((current_cpu == "arm" || current_cpu == "arm64") &&
-    !(is_win && !is_clang)) {
+    !(is_win && !is_clang) && !is_ohos) {
   # TODO(richard.townsend@arm.com): Optimizations temporarily disabled for
   # Windows on Arm MSVC builds, see http://crbug.com/v8/10012.
   if (arm_use_neon) {
@@ -78,7 +78,7 @@ if ((current_cpu == "arm" || current_cpu == "arm64") &&
 }
 
 use_x86_x64_optimizations =
-    (current_cpu == "x86" || current_cpu == "x64") && !is_ios
+    (current_cpu == "x86" || current_cpu == "x64") && !is_ios && !is_ohos
 
 config("zlib_adler32_simd_config") {
   if (use_x86_x64_optimizations) {
@@ -128,7 +128,7 @@ if (use_arm_neon_optimizations) {
     defines = [ "CRC32_ARMV8_CRC32" ]
     if (is_android) {
       defines += [ "ARMV8_OS_ANDROID" ]
-    } else if (is_linux || is_chromeos) {
+    } else if (is_linux || is_chromeos || is_ohos) {
       defines += [ "ARMV8_OS_LINUX" ]
     } else if (is_mac) {
       defines += [ "ARMV8_OS_MACOS" ]
