diff --git a/BUILD.gn b/BUILD.gn
index 8ed0807..14b3570 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -123,11 +123,11 @@ source_set("zlib_adler32_simd") {
 }
 
 if (use_arm_neon_optimizations) {
-  config("zlib_arm_crc32_config") {
+  config(":zlib_arm_crc32_config") {
     defines = [ "CRC32_ARMV8_CRC32" ]
     if (is_android) {
       defines += [ "ARMV8_OS_ANDROID" ]
-    } else if (is_linux || is_chromeos) {
+    } else if (is_linux || is_chromeos || is_ohos) {
       defines += [ "ARMV8_OS_LINUX" ]
     } else if (is_mac) {
       defines += [ "ARMV8_OS_MACOS" ]
