diff --git a/BUILD.gn b/BUILD.gn
index a91d61c8c26..3637e9726c8 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -650,6 +650,20 @@ config("internal_config_base") {
   ]
 }
 
+v8_source_set("wrap_symbols") {
+  sources = [
+    "src/base/wrap_symbols.cc",
+  ]
+
+  public_configs = []
+
+  public_deps = []
+
+  deps = []
+  
+  configs = [ ":internal_config" ]
+}
+
 config("internal_config") {
   defines = []
 
@@ -5204,6 +5218,7 @@ v8_source_set("v8_base_without_compiler") {
     ":v8_tracing",
     ":v8_version",
     "src/inspector:inspector",
+    ":wrap_symbols",
   ]
 
   public_deps = [
@@ -5355,6 +5370,7 @@ v8_source_set("torque_base") {
   deps = [
     ":v8_flags",
     ":v8_shared_internal_headers",
+    ":wrap_symbols",
   ]
 
   public_deps = [ ":v8_libbase" ]
@@ -5541,7 +5557,7 @@ v8_component("v8_libbase") {
 
   public_configs = [ ":libbase_config" ]
 
-  deps = [ ":v8_config_headers" ]
+  deps = [ ":v8_config_headers" , ":wrap_symbols"]
 
   data = []
 
@@ -5725,6 +5741,7 @@ v8_component("v8_libplatform") {
     ":v8_config_headers",
     ":v8_libbase",
     ":v8_tracing",
+    ":wrap_symbols",
   ]
 
   if (v8_use_perfetto) {
diff --git a/src/base/wrap_symbols.cc b/src/base/wrap_symbols.cc
new file mode 100644
index 00000000000..986bf64dd1a
--- /dev/null
+++ b/src/base/wrap_symbols.cc
@@ -0,0 +1,50 @@
+#include <stdlib.h>
+#include "logging.h"
+
+extern "C" {
+
+void* __puerts_wrap__Znwm(unsigned long size) {
+    void* ptr = ::malloc(size);
+    if (!ptr) {
+        FATAL("Fatal process out of memory for new");
+        UNREACHABLE();
+    }
+    return ptr;
+}
+
+void __puerts_wrap__ZdlPv(void* ptr) noexcept {
+    ::free(ptr);
+}
+
+void* __puerts_wrap__Znam(unsigned long size) {
+    void* ptr = ::malloc(size);
+    if (!ptr) {
+        FATAL("Fatal process out of memory for new[]");
+        UNREACHABLE();
+    }
+    return ptr;
+}
+
+void __puerts_wrap__ZdaPv(void* ptr) noexcept {
+    ::free(ptr);
+}
+
+void* __puerts_wrap__ZnwmRKSt9nothrow_t(unsigned long size) {
+    void* ptr = ::malloc(size);
+    if (!ptr) {
+        FATAL("Fatal process out of memory for new (std::nothrow)");
+        UNREACHABLE();
+    }
+    return ptr;
+}
+
+void* __puerts_wrap__ZnamRKSt9nothrow_t(unsigned long size) {
+    void* ptr = ::malloc(size);
+    if (!ptr) {
+        FATAL("Fatal process out of memory for new (std::nothrow)[]");
+        UNREACHABLE();
+    }
+    return ptr;
+}
+
+}
